### DISCUSSION TITLE
how can I remove the media type from the main response

### DISCUSSION THREAD

**Discussion by mdbug** [fastapi/fastapi#7961] [Questions] [answered]
**Description**

How can I remove the media type from the main response?

```
@router.delete("/topics/{topic}",
               summary="Delete topic",
               response_class=None,
               responses={
                   200: {"description": "Topic successfully deleted"},
                   404: {"description": "Topic not found"}
               })
async def delete_topic(topic):
    """
    Delete a topic with all of it's entries
    """
    deleted = await delete_topic(topic)
    if not deleted:
        return Response(status_code=404)

    return
```

In this example the open-api documentation will show no media type for the 404 response, but it will show application/json for the 200 response. How can I remove that. My response body is always empty.

**Comment by euri10** [comment]
When you `return`, implicitly you `return None` and FastAPI under the hood generates a JSONResponse by default.

Should you want to return nothing you have to explicitly specify it using `Response(status_code=whatever status you want)`

see also https://github.com/tiangolo/fastapi/issues/449 for alternatives, in particular @dmontagu's one using the decorator

**Comment by mdbug** [comment]
Thank you for your answer. But if I do it like this
```
@router.delete("/topics/{topic}",
               summary="Delete topic",
               response_class=None,
               responses={
                   200: {"description": "Topic successfully deleted"},
                   404: {"description": "Topic not found"}
               })
async def delete_topic(topic, redis: Redis = Depends(get_redis)):
    """
    Delete a topic with all of it's entries
    """
    deleted = await redis.delete(get_topic_stream_key(topic))
    if deleted == 0:
        return Response(status_code=404)

    return Response(status_code=200)
```
the documentaion still shows
![response](https://user-images.githubusercontent.com/22920400/67842790-bd82b400-faf2-11e9-9752-c6b6749a230c.png)

And if I change `response_class` to `response_class=Response` I get "AssertionError: A response class with media_type is needed to generate OpenAPI"

**Comment by danielgtaylor** [comment]
Confirmed. The OpenAPI output is broken for 204 responses and the workaround mentioned above no longer works to return a blank body. Looks like a bug to me.

**Comment by tiangolo** [comment, marked as answer]
You now can use `response_class=Response`, I'm not sure since which version :shrug: (we are now on version `0.54.1`).

But this seems to work as you expected:

```Python
from fastapi import FastAPI, Response

app = FastAPI()


@app.delete(
    "/topics/{topic}",
    summary="Delete topic",
    response_class=Response,
    responses={
        200: {"description": "Topic successfully deleted"},
        404: {"description": "Topic not found"},
    },
)
async def delete_topic(topic: str):
    """
    Delete a topic with all of it's entries
    """
    return Response(status_code=200)
```

**Comment by mdbug** [comment]
thanks
